simple-dns/src/dns/character_string.rs

```diff
@@ -37,6 +36,12 @@ impl<'a> CharacterString<'a> {
     }
 }
 
+impl<'a> From<CharacterString<'a>> for String {
+    fn from(val: CharacterString<'a>) -> Self {
+        String::from_utf8(val.data.into_owned()).unwrap()
+    }
+}
+
```

Comment:
Can you change this to a `TryFrom` implementaiton? Since it can be called in a CharacterString that is not valid utf8

Reply:
done